### PR TITLE
Adding option to remove sites from auto-contribute

### PR DESCRIPTION
**Pull request by ryanml**
Addresses: https://github.com/brave/brave-browser/issues/967

Native ledger work:
https://github.com/brave-intl/bat-native-ledger/pull/87
https://github.com/brave-intl/bat-native-ledger/pull/93

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Needed or QA/No-QA-Needed) to include the closed issue in milestone 

## Test Plan:
1. Launch Brave, enable rewards
2. Visit 6 sites to populate contributions table
3. In the contribution table, click `See all ${n} sites` to display the removal modal
4. Click the `x` icon next to a few publishers
5. Confirm they are excluded and removed from the contribution table

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by ryanml** [comment]
Note to reviewers:

I'm not sure how it fits in with the planned UX but I'd like for there to be the exclusion option present when there are less than 6 sites, currently there needs to be at least that amount for option to be there. This can be addressed in another pr over on `brave-ui`

**Comment by jasonrsadler** [comment]
master: 7564b7c
0.55.x: d8716b05

**Comment by bbondy** [comment]
For next time pls also add `0.55.x` here as an indication that it's the first channel it appears on. I review a list of PRs periodically to make sure we didn't forget to uplift some things.  I added it this time 👍

File: components/definitions/rewards.d.ts [hunk 1]
```diff
@@ -82,10 +82,13 @@ declare namespace Rewards {
     expiryTime: number
   }
 
+  export type Excluded = 0 | 1 | 2
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply components/definitions/rewards.d.ts]: @ryanml

**Comment by ryanml** [comment]
@nejczdovc I need to be able to show the number of excluded sites for contribute modals restore option, if we fetch only the included publishers, I won't be able to calculate the difference. What do you suggest? Fetching only the count?

**Comment by ryanml** [comment]
@nejczdovc - you want to see something like:

`export type Excluded = 'DEFAULT' | 'EXCLUDED' | 'INCLUDED'` ?

**Comment by NejcZdovc** [comment]
@ryanml something like this

```
export enum Excluded {
    DEFAULT = 0,
    INCLUDED = 1,
    EXCLUDED = 2
}
```

**Pull request merged**